File: modified_scripts/c/binary_search/gemma_2b.c
int binary_search(int arr[], int n, int target) {
  int left = 0;
  int right = n - 1;

  while (left <= right) {
    int mid = (left + right) / 2;

    if (arr[mid] == target) {
      return mid;
    } else if (arr[mid] < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return -1;
}

File: modified_scripts/r/binary_search/gemma_2b.r

binary_search <- function(arr, target) {
  left <- 1
  right <- length(arr)
  while (left <= right) {
    mid <- floor((left + right) / 2)
    guess <- arr[mid]
    if (guess == target) {
      return(mid)
    } else if (guess < target) {
      left <- mid + 1
    } else {
      right <- mid - 1
    }
  }
  return(-1)
}